### ISSUE TITLE
EclipsePlugin-tests plugin classpath is broken

### ISSUE THREAD

**Issue by iloveeclipse** [spotbugs/spotbugs#160]
It was broken via #40 .
eclipsePlugin-tests .classpath is wrong, developing tests in Eclipse is not possible anymore.

Also after fixing the main problem with broken .classpath, MANIFEST.MF must be updated to change dependency from old findbugs plugin to the new id.

**Comment by KengoTODA** [comment]
Current status: I found that we should do following to fix.

* `spotbugs` and `spotbugs-annotation` should follow OSGI and export their classes to users
    * `spotbugs-annotation` depends on JSR305, and need to export packages in it
    * Both of them need to copy dependencies into `lib` directory in project (we can ignore them by `.gitignore`)

## MANIFEST.MF in jsr-305

```
Manifest-Version: 1.0
Bundle-Description: JSR305 Annotations for Findbugs
Bundle-License: http://www.apache.org/licenses/LICENSE-2.0.txt
Bundle-SymbolicName: org.jsr-305
Archiver-Version: Plexus Archiver
Built-By: lan
Bnd-LastModified: 1490936130302
Bundle-ManifestVersion: 2
Tool: Bnd-2.1.0.20130426-122213
Export-Package: javax.annotation;uses:="javax.annotation.meta";version
 ="3.0.2",javax.annotation.concurrent;version="3.0.2",javax.annotation
 .meta;uses:="javax.annotation";version="3.0.2"
Bundle-Name: FindBugs-jsr305
Bundle-Version: 3.0.2
Created-By: Apache Maven Bundle Plugin
Build-Jdk: 1.8.0_101
```

**Comment by iloveeclipse** [comment]
> Both of them need to copy dependencies into lib directory in project
By dependencies you mean probably all the libraries they have specified on the classpath. Yes. This (manifests and dependencies in lib/) would restore one part of previous state.

Another issue is that .classpath files for plugin projects aren't properly generated - they seem to be generated for plain Java projects, but they should be generated for OSGI bundles (aka Eclipse plugins).

**Comment by KengoTODA** [comment]
@jsotuyod It seems that we need to use OSGI plugin to generate proper manifest file.

This https://github.com/KengoTODA/spotbugs/commit/d490eb0cca07cdd5de09e4fd5162939b79e8f8e2 is my commit to update `spotbugs-annotation` project, however generated Eclipse project seems still invalid.
